sites-enabled/24-chat.freecodecamp.dev.conf

```diff
@@ -0,0 +1,42 @@
+# ---------------------------
+# chat.freecodecamp.dev
+# ---------------------------
+
+server {
+  listen 443 ssl http2;
+  listen [::]:443 ssl http2;
+
+  server_name chat.freecodecamp.dev;
+
+  # SSL
+  include snippets/common/ssl-freecodecamp-dev.conf;
+
+  # security
+  include snippets/common/security.conf;
+
+  # reverse proxy chat
+  location / {
+    if (-f /etc/nginx/maintenance/CHAT.txt) {
+      return 503;
+    }
+
+    # do not index, follow Chat applications
+    add_header  X-Robots-Tag "noindex, nofollow, nosnippet, noarchive";
+
+    proxy_pass http://chat-dev;
+    include snippets/common/proxy-params.conf;
+  }
+
+  # additional config
+  include snippets/common/general.conf;
+}
+
+# HTTP redirect
+server {
+  listen 80;
+  listen [::]:80;
+
+  server_name chat.freecodecamp.dev;
+
+  return 301 https://chat.freecodecamp.dev$request_uri;
+}
```

Comment:
This bit is technically useless because .dev domains mandate browsers to use HSTS. But I am keeping it here for sake of posterity (and does not hurt)